### ISSUE TITLE
Requirements to finetune / train?

### ISSUE THREAD

**Issue by fire** [seal-rg/recurrent-pretraining#3]
Let's say we have a data set like https://huggingface.co/datasets/AI-MO/NuminaMath-1.5.

> This model was trained on 21 segments of 4096 AMD MI-250X GPUs on the OLCF Frontier Supercomputer in early December 2024.

Will we be able to train on lesser hardware?

**Comment by JonasGeiping** [comment]
Hi! Yeah, post-training on a dataset like that would not require this much compute. The model's footprint is actually relatively small at train-time, especially when using gradient checkpointing at the iteration level, so you should be able to fit it onto smaller cards as well for finetuning (but maybe not so easily with the code from this repo, or rather without changing it).

**Comment by fire** [comment]
Any estimates on a rtx 4090?

**Comment by JonasGeiping** [comment]
No, not really - but really the bigger question that model speed will be how much finetuning you intend to do.

**Comment by rawwerks** [comment]
@JonasGeiping - any guidelines on finetuning huginn-0125?

**Comment by JonasGeiping** [comment]
No, not yet. You could finetune it like a normal transformer model (but probably best with the same randomized depth objective as used during pretraining). However, I imagine there will be better ways to finetune the model, but that is a research question that we are currently investigating.

**Comment by rawwerks** [comment]
ok thanks. even if not "optimal" or even "academically interesting", any small example code on how to set up fine-tuning of your model would be greatly appreciated!

**Comment by JonasGeiping** [comment]
To be clear :) - This is a new model, no one (to my knowledge) has finetuned this model yet, or even written code to do so. 

The existing huggingface implementation of the model might be sufficient to support finetuning with the huggingface trainer, but this is untested.
